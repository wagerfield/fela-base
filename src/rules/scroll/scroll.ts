export interface ScrollRuleOptions {
  scrolling?: any
  x?: boolean
  y?: boolean
}

export default ({
  scrolling = "touch",
  x = false,
  y = true
}: ScrollRuleOptions = {}) => ({
  overflowX: x ? "auto" : "hidden",
  overflowY: y ? "auto" : "hidden",
  scrolling
})
