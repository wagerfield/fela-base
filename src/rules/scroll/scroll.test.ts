import scroll from "./scroll"

describe("scroll(props)", () => {
  it("renders expected snapshots", () => {
    expect(scroll()).toMatchSnapshot()
    expect(
      scroll({
        x: true
      })
    ).toMatchSnapshot()
  })
})
