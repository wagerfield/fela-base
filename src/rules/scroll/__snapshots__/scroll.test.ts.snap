// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scroll(props) renders expected snapshots 1`] = `
Object {
  "overflowX": "hidden",
  "overflowY": "auto",
  "scrolling": "touch",
}
`;

exports[`scroll(props) renders expected snapshots 2`] = `
Object {
  "overflowX": "auto",
  "overflowY": "auto",
  "scrolling": "touch",
}
`;
